### ISSUE TITLE
dynamic_axes in torch.onnx.export seems broken in 2.9.0

### ISSUE THREAD

**Issue by wdhongtw** [pytorch/pytorch#165748]
### 🐛 Describe the bug

`dynamic_axes` argument of `torch.onnx.export` not functional in latest release v2.9.0.

Using the sample code from https://docs.pytorch.org/docs/stable/onnx.html

```python
import torch
import onnx

class SumModule(torch.nn.Module):
    def forward(self, x):
        return torch.sum(x, dim=1)


torch.onnx.export(
    SumModule(),
    (torch.ones(2, 2),),
    "onnx.pb",
    input_names=["x"],
    output_names=["sum"],
    dynamic_axes={
        "x": {0: "my_custom_axis_name"},
        "sum": [0],
    },
)

onnx_model = onnx.load("onnx.pb")
print(onnx_model.graph.input)
print(onnx_model.graph.output)
```

in 2.9.0 dynamic axis names are some serial number like `s77`

```text
[name: "x"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_param: "s77"
      }
      dim {
        dim_value: 2
      }
    }
  }
}
metadata_props {
  key: "pkg.torch.export.graph_signature.InputSpec.kind"
  value: "USER_INPUT"
}
metadata_props {
  key: "pkg.torch.export.graph_signature.InputSpec.persistent"
  value: "None"
}
metadata_props {
  key: "pkg.torch.onnx.original_node_name"
  value: "x"
}
]
[name: "sum"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_param: "s77"
      }
    }
  }
}
metadata_props {
  key: "pkg.torch.export.graph_signature.OutputSpec.kind"
  value: "USER_OUTPUT"
}
metadata_props {
  key: "pkg.torch.onnx.original_node_name"
  value: "sum_1"
}
]
```

While in 2.8.0, the dynamic axis names are as expected.

```text
[name: "x"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_param: "my_custom_axis_name"
      }
      dim {
        dim_value: 2
      }
    }
  }
}
]
[name: "sum"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_param: "sum_dynamic_axes_1"
      }
    }
  }
}
]
```

### Versions

Collecting environment information...
PyTorch version: 2.9.0+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.12.12 (main, Oct 10 2025, 08:52:57) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.8.0-1015-gcp-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               180
On-line CPU(s) list:                  0-179
Vendor ID:                            AuthenticAMD
Model name:                           AMD EPYC 9B14
CPU family:                           25
Model:                                17
Thread(s) per core:                   1
Core(s) per socket:                   90
Socket(s):                            2
Stepping:                             1
BogoMIPS:                             5200.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext ssbd ibrs ibpb stibp vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx512_bf16 clzero xsaveerptr wbnoinvd arat avx512vbmi umip avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid fsrm
Hypervisor vendor:                    KVM
Virtualization type:                  full
L1d cache:                            5.6 MiB (180 instances)
L1i cache:                            5.6 MiB (180 instances)
L2 cache:                             180 MiB (180 instances)
L3 cache:                             768 MiB (24 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-89
NUMA node1 CPU(s):                    90-179
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Mitigation; Safe RET
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP disabled; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] Could not collect
[conda] Could not collect

cc @justinchuby @titaiwangms

**Comment by justinchuby** [comment]
Was there renaming done for that? Or no

**Comment by justinchuby** [comment]
>  I want to remove this favor (axes to shapes conversion) to users though.

Maybe we can deprecate first to allow a transition process.

**Comment by wdhongtw** [comment]
@titaiwangms Thanks for the timely and clear response. 😀
